lib/rubocop/cop/lint/constant_defined_in_block.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Do not define constants within a block, since the block's scope does not
+      # isolate or namespace the constant in any way.
+      #
+      # Define the constant outside of the block instead, or use a variable or
+      # method if defining the constant in the outer scope would be problematic.
+      #
+      # @example
+      #   # bad
+      #   task :lint do
+      #     FILES_TO_LINT = Dir['lib/*.rb']
+      #   end
+      #
+      #   # bad
+      #   describe 'making a request' do
+      #     class TestRequest; end
+      #   end
+      #
+      #   # good
+      #   task :lint do
+      #     files_to_lint = Dir['lib/*.rb']
+      #   end
+      #
+      #   # good
+      #   describe 'making a request' do
+      #     let(:test_request) { Class.new }
+      #   end
+      class ConstantDefinedInBlock < Base
```

Comment:
Probably we should name this `ConstantDefinitionInBlock` to be more consistent with existing cop names.

Reply:
Good call, I updated the name.